pkg/cache/v3/resource.go

```diff
@@ -202,22 +202,11 @@ func getListenerReferences(src *listener.Listener, out map[resource.Type]map[str
 
 	// Extract route configuration names from HTTP connection manager.
 	for _, chain := range src.GetFilterChains() {
-		for _, filter := range chain.GetFilters() {
-			config := resource.GetHTTPConnectionManager(filter)
-			if config == nil {
-				continue
-			}
-
-			// If we are using RDS, add the referenced the route name.
-			if name := config.GetRds().GetRouteConfigName(); name != "" {
-				routes[name] = true
-			}
+		getListenerReferencesFromChain(chain, routes)
+	}
 
-			// If the scoped route mapping is embedded, add the referenced route resource names.
-			for _, s := range config.GetScopedRoutes().GetScopedRouteConfigurationsList().GetScopedRouteConfigurations() {
-				routes[s.GetRouteConfigurationName()] = true
-			}
-		}
+	if src.DefaultFilterChain != nil {
```

Comment:
Minor: the current code does not make an assumption on src not being `nil`, whereas this change would panic. Can you use the protobuf safe getter for consistency?

Reply:
Ah, good point. Fixed.